02_bubble/runnerdave/main.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, bubble([]int{3, 2, 1, 5}))
+}
+
+func bubble(s []int) []int {
+	len := len(s)
+	for i := 0; i < len-1; i++ {
+		for j := 0; j < len-i-1; j++ {
+			if s[j] > s[j+1] {
+				temp := s[j]
```

Comment:
You can do this in one line using multiple assignment...
`a, b = b, a`
This doesn't lose one of the values as you might expect it to in other languages.

Reply:
this is very nice, thanks for the detailed review